File: lc_p88_s2.py
from typing import *


class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        current1 = m - 1
        current2 = n - 1
        back = len(nums1) - 1

        while current1 >= 0 and current2 >= 0:
            if nums1[current1] >= nums2[current2]:
                nums1[back] = nums1[current1]
                back -= 1
                current1 -= 1
            else:
                nums1[back] = nums2[current2]
                back -= 1
                current2 -= 1

        while current2 >= 0:
            nums1[back] = nums2[current2]
            back -= 1
            current2 -= 1


def main() -> None:
    nums1: List[int] = [1, 2, 3, 0, 0, 0]
    nums2: List[int] = [2, 5, 6]
    Solution().merge(nums1, 3, nums2, 3)
    print(nums1)


if __name__ == "__main__":
    main()

File: lc_p88_s2.c
#include <stdio.h>
#include <stdlib.h>

void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n) {
  int current1 = m - 1;
  int current2 = n - 1;
  int back = nums1Size - 1;

  while (current1 >= 0 && current2 >= 0) {
    if (nums1[current1] >= nums2[current2]) {
      nums1[back--] = nums1[current1--];
    } else {
      nums1[back--] = nums2[current2--];
    }
  }

  while (current2 >= 0) {
    nums1[back--] = nums2[current2--];
  }
}

int main(void) {
    int nums1[6] = {1, 2, 3, 0, 0, 0};
    int nums2[3] =  {2, 5, 6};
    merge(nums1, 6, 3, nums2, 3, 3);

    for (int i = 0; i < 6; i++) {
        printf("%d ", nums1[i]);
    }

    return 0;
}
